updater/lib/dependabot/updater/operations/group_update_all_versions.rb

```diff
@@ -21,9 +21,10 @@ class GroupUpdateAllVersions
         include GroupUpdateCreation
 
         def self.applies_to?(job:)
-          return false if job.security_updates_only?
           return false if job.updating_a_pull_request?
-          return false if job.dependencies&.any?
+          if Dependabot::Experiments.enabled?(:grouped_security_updates_disabled) && job.security_updates_only?
+            return false
+          end
```

Comment:
this is a bit hard to read; can it be simplified with an `unless` check (maybe?)

Reply:
I can't tell how `unless` would make this less confusing.

This flag was added in https://github.com/dependabot/dependabot-core/pull/8529, I'm only moving it over in this PR. We could add an experiment and turn it on which will clean this logic up. I can do that in a followup PR.